src/riverroute/MOSART_physics_mod.F90

```diff
@@ -1,710 +1,730 @@
+      ! part 3 is a rectagular with the width rwid0
+
+      ! Arguments
+      real(r8), intent(in) :: mr_, rwidth_, rwidth0_, rdepth_ ! wetted channel area, channel width, flood plain wid, water depth
+      real(r8)             :: hr_                             ! water depth
+
+      real(r8) :: SLOPE1  ! slope of flood plain, TO DO
+      real(r8) :: deltamr_
+
+      SLOPE1 = SLOPE1def
+      if(mr_ <= TINYVALUE) then
+         hr_ = 0._r8
+      else
+         if(mr_ - rdepth_*rwidth_ <= TINYVALUE) then ! not flooded
+            hr_ = mr_/rwidth_
+         else ! if flooded, the find out the equivalent depth
+            if(mr_ > rdepth_*rwidth_ + (rwidth_ + rwidth0_)*SLOPE1*((rwidth0_-rwidth_)/2._r8)/2._r8 + TINYVALUE) then
+               deltamr_ = mr_ - rdepth_*rwidth_ - (rwidth_ + rwidth0_)*SLOPE1*((rwidth0_ - rwidth_)/2._r8)/2._r8;
+               hr_ = rdepth_ + SLOPE1*((rwidth0_ - rwidth_)/2._r8) + deltamr_/(rwidth0_);
+            else
+               deltamr_ = mr_ - rdepth_*rwidth_;
+               !           !hr_ = rdepth_ + (-rwidth_+sqrt( rwidth_**2._r8  +4._r8*deltamr_/SLOPE1))*SLOPE1/2._r8
+               hr_ = rdepth_ + (-rwidth_+sqrt((rwidth_*rwidth_)+4._r8*deltamr_/SLOPE1))*SLOPE1/2._r8
+            end if
+         end if
+      end if
+      return
+   end function GRHR
+
+   !-----------------------------------------------------------------------
+
+   function GRPR(hr_, rwidth_, rwidth0_,rdepth_) result(pr_)
+      ! Function for estimating maximum water depth assuming rectangular channel and tropezoidal flood plain
+      ! here assuming the channel cross-section consists of three parts, from bottom to up,
+      ! part 1 is a rectangular with bankfull depth (rdep) and bankfull width (rwid)
+      ! part 2 is a tropezoidal, bottom width rwid and top width rwid0, height 0.1*((rwid0-rwid)/2), assuming slope is 0.1
+      ! part 3 is a rectagular with the width rwid0
+
+      ! Arguments
+      real(r8), intent(in) :: hr_, rwidth_, rwidth0_, rdepth_ ! wwater depth, channel width, flood plain wid, water depth
+      real(r8)             :: pr_                             ! water depth
+
+      real(r8) :: SLOPE1  ! slope of flood plain, TO DO
+      real(r8) :: deltahr_
+      logical, save :: first_call = .true.
+
+      SLOPE1 = SLOPE1def
+      if (first_call) then
+         sinatanSLOPE1defr = 1.0_r8/(sin(atan(SLOPE1def)))
+      endif
+      first_call = .false.
+
+      if(hr_ < TINYVALUE) then
+         pr_ = 0._r8
+      else
+         if(hr_ <= rdepth_ + TINYVALUE) then ! not flooded
+            pr_ = rwidth_ + 2._r8*hr_
+         else
+            if(hr_ > rdepth_ + ((rwidth0_-rwidth_)/2._r8)*SLOPE1 + TINYVALUE) then
+               deltahr_ = hr_ - rdepth_ - ((rwidth0_-rwidth_)/2._r8)*SLOPE1
+               ! pr_ = rwidth_ + 2._r8*(rdepth_ + ((rwidth0_-rwidth_)/2._r8)*SLOPE1/sin(atan(SLOPE1)) + deltahr_)
```

Comment:
Again I suggest we remove code instead of leaving it in comments.

Reply:
Totally agree!